JetBrains/compose-multiplatform-core
# Temporary move `rememberComposableLambda` from commonMain to jvmAndNativeMain

**Description:**
Currently, compose compiler plugin doesn't handle properly this function when targeting web.

We will move the function back to commonMain, after all compiler plugins support the new function in web targets.


Fixes CMP-1547 for desktop and k/native. Web is still affected by the issue.

## Testing
1. Open :compose:mpp:demo on desktop
2. Go to Components -> Dialogs -> Dialog with text field
3. Close the dialog and Go back (press the back button)
No unexpected dialog is shown.

To reproduce the issue, comment `rememberComposableLambda`.
Repeat the steps above.
An unexpected dialog will be shown after back button is pressed.



**Note**: when we move the function back, it means that CM 1.7 projects won't be able to use kotlin 2.0 with web targets. Only kotlin 2.0.10 if the compiler fix is accepted.
